### PR TITLE
Added scheduled-jobs-alert.tf

### PR DESCRIPTION
**Pull request by apereverzin**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/BPS-1019

### Change description ###
Added scheduled-jobs-alert.tf

This PR is a 'reference implementation', similar to heartbeat alert in bulk-scan-shared-infrastructure, actual values for frequency and time window are subject to discussion.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: scheduled-jobs-alert.tf [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  app_insights_name = "${azurerm_application_insights.appinsights.name}"
+
+  enabled    = "${var.env == "prod"}"
+  alert_name = "Reform_Blob_Dispatcher_-_BSP"
+  alert_desc = "Triggers when no logs from blob-dispatcher job found within timeframe."
+
+  app_insights_query = "traces | where message startswith 'Started blob-dispatcher job'"
+
+  frequency_in_minutes       = 10
+  time_window_in_minutes     = 10
```
Comment (lgonczar) [inline scheduled-jobs-alert.tf]: Probably best to change it to something like 30 minutes maybe (definitely not less than 15). There's a 10 minute delay between consecutive job runs, so there's an even greater gap between start times.

File: scheduled-jobs-alert.tf [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  app_insights_name = "${azurerm_application_insights.appinsights.name}"
+
+  enabled    = "${var.env == "prod"}"
+  alert_name = "Reform_Handle_Rejected_Files_-_BSP"
+  alert_desc = "Triggers when no logs from handle-rejected-files job found within timeframe."
+
+  app_insights_query = "traces | where message startswith 'Started handle-rejected-files job'"
+
+  frequency_in_minutes       = 120
+  time_window_in_minutes     = 120
```
Comment (lgonczar) [inline scheduled-jobs-alert.tf]: Does this job run only once per day?

Comment (apereverzin) [reply scheduled-jobs-alert.tf]: I assume '120' means every 120 minutes i.e. 2 hrs

Comment (lgonczar) [reply scheduled-jobs-alert.tf]: I've just checked. The job only runs once per day, so this alert will be triggered every two hours.

Comment (apereverzin) [reply scheduled-jobs-alert.tf]: Have made it 2160 minutes - 1 day and a half

File: scheduled-jobs-alert.tf [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  app_insights_name = "${azurerm_application_insights.appinsights.name}"
+
+  enabled    = "${var.env == "prod"}"
+  alert_name = "Reform_Handle_Rejected_Files_-_BSP"
+  alert_desc = "Triggers when no logs from handle-rejected-files job found within timeframe."
+
+  app_insights_query = "traces | where message startswith 'Started handle-rejected-files job'"
+
+  frequency_in_minutes       = 2160
+  time_window_in_minutes     = 2160
```
Comment (rkondratowicz) [inline scheduled-jobs-alert.tf]: a comment on how many hours this is will be appreciated.
also a comment explaining why so rarely.

File: scheduled-jobs-alert.tf [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  alert_name = "Reform_Blob_Dispatcher_-_BSP"
+  alert_desc = "Triggers when no logs from blob-dispatcher job found within timeframe."
+
+  app_insights_query = "traces | where message startswith 'Started blob-dispatcher job'"
+
+  frequency_in_minutes       = 30
+  time_window_in_minutes     = 30
+  severity_level             = "1"
+  action_group_name          = "${module.alert-action-group.action_group_name}"
+  custom_email_subject       = "Reform Scan blob-dispatcher"
```
Comment (rkondratowicz) [inline scheduled-jobs-alert.tf]: it could be much more specific.

File: scheduled-jobs-alert.tf (hunk 2 above)
Comment (apereverzin) [reply scheduled-jobs-alert.tf]: It looks like there is no way to make schedule of this alert (Reform_Handle_Rejected_Files_-_BSP) more specific e.g. linked to the time of running handle-rejected-files job - cnp-module-metric-alert module provides only 2 time related parameters: frequency_in_minutes and  time_window_in_minutes.

File: scheduled-jobs-alert.tf (hunk 3 above)
Comment (lgonczar) [reply scheduled-jobs-alert.tf]: How about setting the frequency to sth like 1 hour (can be longer) and time window to 25 hours? This way we would be notified much quicker.

Comment (apereverzin) [reply scheduled-jobs-alert.tf]: Have done this

**Comment by lgonczar** [comment]
Looks good to me. Maybe we could also change the names of these alerts from something like this:
```
Reform_Blob_Dispatcher_-_BSP
```
to sth like:

```
Blob dispatcher job not running
```

Turns out these can just be user friendly names.

**Comment by doncem** [comment]
> Looks good to me. Maybe we could also change the names of these alerts from something like this:
> 
> ```
> Reform_Blob_Dispatcher_-_BSP
> ```
> 
> to sth like:
> 
> ```
> Blob dispatcher job not running
> ```
> 
> Turns out these can just be user friendly names.

names are restricted format and spaces are not allowed anymore since (i don't remember actually)...

**Comment by lgonczar** [comment]
Oh, I see. I saw an alert with spaces in prod and thought it was allowed. If spaces don't work, we could replace them with underscores or hyphens. And we should drop the `Reform` word at the front, because it doesn't add any information. All these alerts live in the scope of reform-scan-{env} App Insights, so we already know it's about Reform Scan.

File: scheduled-jobs-alert.tf [hunk 5]
```diff
@@ -0,0 +1,65 @@
+module "blob-dispatcher-alert" {
+  source            = "git@github.com:hmcts/cnp-module-metric-alert"
+  location          = "${azurerm_application_insights.appinsights.location}"
+  app_insights_name = "${azurerm_application_insights.appinsights.name}"
+
+  enabled    = "${var.env == "prod"}"
+  alert_name = "Reform Blob dispatcher job not running"
```
Comment (apereverzin) [inline scheduled-jobs-alert.tf]: The build has passed with the names with spaces. When the restriction mentioned will take effect? @doncem

Comment (doncem) [reply scheduled-jobs-alert.tf]: PR doesn't run terraform code 😉

**Pull request merged**